### PR TITLE
#30712 - Product grid for category page

### PR DESCRIPTION
**Pull request by anqaka**
https://dev.snowdog.pro/issues/30712

### CODE REVIEW

File: web/images/icons-sprite.svg [hunk 1]
```diff
@@ -0,0 +1 @@
...753 2.24 25.89 6.71 39.398 4.472 13.514 9.567 25.03 15.276 34.546 5.708 9.514 12.18 18.792 19.414 27.834 7.233 9.04 12.52 15.272 15.846 18.698 3.33 3.426 5.948 5.903 7.85 7.427L243.25 469.938c3.427 3.426 7.614 5.144 12.562 5.144s9.138-1.718 12.563-5.144l177.87-171.31c43.588-43.58 65.38-86.406 65.38-128.472 0-41.877-12.085-74.61-36.26-98.207zm-53.96 199.845L255.812 431.39 89.938 271.508c-35.594-35.585-53.388-69.374-53.388-101.35 0-15.416 2.046-29.027 6.136-40.825 4.093-11.8 9.327-21.177 15.703-28.124 6.376-6.95 14.13-12.607 23.267-16.988 9.14-4.377 18.086-7.328 26.84-8.85 8.754-1.52 18.08-2.28 27.978-2.28 9.896 0 20.557 2.423 31.977 7.278 11.418 4.853 21.934 10.944 31.545 18.27 9.613 7.333 17.845 14.184 24.7 20.558 6.85 6.38 12.56 12.23 17.128 17.56 3.424 4.188 8.09 6.282 13.99 6.282s10.56-2.094 13.99-6.283c4.567-5.33 10.28-11.182 17.13-17.56 6.852-6.373 15.085-13.22 24.694-20.556 9.612-7.328 20.128-13.42 31.552-18.272 11.416-4.854 22.08-7.28 31.977-7.28s19.218.762 27.976 2.282c8.757 1.52 17.702 4.473 26.84 8.85 9.137 4.38 16.892 10.042 23.267 16.988 6.376 6.947 11.612 16.324 15.705 28.124 4.086 11.798 6.132 25.41 6.132 40.824-.002 31.977-17.89 65.86-53.675 101.64z"/></symbol></svg>
```
Comment (Igloczek) [inline web/images/icons-sprite.svg]: Shouldn't be added to the repository.

File: styles/modules/_product.scss [hunk 2]
```diff
@@ -0,0 +1,60 @@
+.product {
```
Comment (Igloczek) [inline styles/modules/_product.scss]: All of the product-ish things should be moved to `Magento_Catalog` module dir.

File: styles/modules/_button.scss [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.button {
+    align-items: center;
+
+    &--wide {
+        width: 100%;
```
Comment (Igloczek) [inline styles/modules/_button.scss]: Needs to be moved to the components.

File: styles/styles.scss [hunk 4]
```diff
@@ -40,5 +40,11 @@
 @import '../Snowdog_AlpacaComponents/02-elements/tab/tab';
 
 // alpaca components' modules
-@import '../Snowdog_AlpacaComponents/03-modules/products-grid/products-grid';
+// @import '../Snowdog_AlpacaComponents/03-modules/products-grid/products-grid';
```
Comment (Igloczek) [inline styles/styles.scss]: ☠️  dead code

File: web/images/icons-sprite.svg (hunk 1 above)
Comment (anqaka) [inline web/images/icons-sprite.svg]: why icon sprite shouldn't be added? It has to be copied from components repository so we need this file inside the theme...

File: styles/modules/_product.scss (hunk 2 above)
Comment (anqaka) [inline styles/modules/_product.scss]: moved to alpaca-components in https://github.com/SnowdogApps/magento2-alpaca-ui-components/pull/43

File: styles/styles.scss (hunk 4 above)
Comment (anqaka) [inline styles/styles.scss]: fixed

**Pull request merged**